src/datasources/perf-ds/PerformanceDataSource.tsx

```diff
@@ -224,4 +243,68 @@ export class PerformanceDataSource extends DataSourceApi<PerformanceQuery> {
 
         return response && response.data ? response.data as OnmsResourceDto : null
     }
+
+    async metricFindLocations() {
+        return await this.simpleRequest.getLocations();
+    }
+
+    async metricFindNodeFilterQuery(query) {
+        let nodes = await this.simpleRequest.getNodesByFilter(query);
+        let results = [] as any[];
+        nodes.forEach(node => {
+            let nodeCriteria = node.id.toString();
+            if (node.foreignId !== null && node.foreignSource !== null) {
+                nodeCriteria = node.foreignSource + ":" + node.foreignId;
+            }
+            results.push({ text: node.label, value: nodeCriteria, expandable: true });
+        });
+        return results;
+    }
+
+    async metricFindNodeResourceQuery(query, ...options) {
```

Comment:
This function might benefit from being broken down into smaller parts. buildNodeResourceOptions(), request, extractResultsFromResponse()

Reply:
I'll include those shcnage in my next PR